VERSION ""


NS_ :
    NS_DESC_
    CM_
    BA_DEF_
    BA_
    VAL_
    CAT_DEF_
    CAT_
    FILTER
    BA_DEF_DEF_
    EV_DATA_
    ENVVAR_DATA_
    SGTYPE_
    SGTYPE_VAL_
    BA_DEF_SGTYPE_
    BA_SGTYPE_
    SIG_TYPE_REF_
    VAL_TABLE_
    SIG_GROUP_
    SIG_VALTYPE_
    SIGTYPE_VALTYPE_
    BO_TX_BU_
    BA_DEF_REL_
    BA_REL_
    BA_DEF_DEF_REL_
    BU_SG_REL_
    BU_EV_REL_
    BU_BO_REL_
    SG_MUL_VAL_

BS_:

BU_: EON XXX INTERCEPTOR ANY ACTUATOR SAFETY

CM_ "KommuActuator CAN messages starts here";

BO_ 512 GAS_COMMAND: 8 SAFETY
 SG_ GAS_COMMAND : 7|16@0+ (1,0) [0|1] "" ACTUATOR
 SG_ GAS_COMMAND2 : 23|16@0+ (1,0) [0|1] "" ACTUATOR
 SG_ ENABLE : 40|1@0+ (1,0) [0|1] "" ACTUATOR
 SG_ COUNTER_PEDAL : 51|4@0+ (1,0) [0|15] "" ACTUATOR
 SG_ CHECKSUM_PEDAL : 63|8@0+ (1,0) [0|255] "" ACTUATOR

BO_ 513 GAS_SENSOR: 6 ACTUATOR
 SG_ STATE : 35|4@0+ (1,0) [0|15] "" SAFETY
 SG_ INTERCEPTOR_GAS : 7|16@0+ (0.0007,-0.342) [0|1] "" SAFETY
 SG_ INTERCEPTOR_GAS2 : 23|16@0+ (0.0007,-0.691) [0|1] "" SAFETY

BO_ 514 TORQUE_COMMAND: 8 SAFETY
 SG_ INTERCEPTOR_MAIN_TORQUE : 7|16@0+ (1,0) [0|1] "" ACTUATOR
 SG_ INTERCEPTOR_SUB_TORQUE : 23|16@0+ (1,0) [0|1] "" ACTUATOR
 SG_ DIRECTION : 32|1@0+ (1,0) [0|1] "" ACTUATOR
 SG_ ENABLE : 40|1@0+ (1,0) [0|1] "" ACTUATOR
 SG_ COUNTER_STEERING : 51|4@0+ (1,0) [0|15] "" ACTUATOR
 SG_ CHECKSUM_STEERING : 63|8@0+ (1,0) [0|255] "" ACTUATOR

CM_ "Perodua Ativa CAN messages starts here";

BO_ 161 BRAKE: 8 XXX
 SG_ SPEED : 39|16@0+ (0.0108,0) [0|65535] "" XXX
 SG_ BRAKE_PRESSURE : 48|8@1+ (0.005532,0) [0|255] "" XXX
 SG_ CHECKSUM : 63|8@0+ (0.00003322,-0.00774026) [0|65535] "" XXX
 SG_ BRAKE_ENGAGED : 5|1@0+ (1,0) [0|1] "" XXX

BO_ 164 STEERING_MODULE: 8 XXX
 SG_ MAIN_TORQUE : 9|12@0- (1,0) [0|65535] "" XXX
 SG_ CHECKSUM : 63|8@0+ (1,0) [0|255] "" XXX
 SG_ STEER_ANGLE : 46|15@0- (0.10047,0) [-500|500] "" XXX

BO_ 398 GAS_PEDAL: 8 XXX
 SG_ PULSE_WHEN_PEDAL_ZERO : 2|1@0+ (1,0) [0|1] "" XXX
 SG_ CHECKSUM : 63|8@0+ (1,0) [0|255] "" XXX
 SG_ APPS_1 : 12|13@0+ (0.00015625,0) [0|2097151] "%" XXX
 SG_ APPS_2 : 28|13@0+ (0.00006742,0) [0|1023] "" XXX
 SG_ APPS_3 : 44|13@0+ (0.00009,0) [0|7] "" XXX

BO_ 399 GAS_PEDAL_2: 8 XXX
 SG_ GAS_PEDAL_STEP : 1|1@0+ (1,0) [0|1] "" XXX
 SG_ ACC_CMD : 23|16@0+ (1,0) [0|65535] "" XXX
 SG_ MAYBE_ACC_BRAKE1 : 39|16@0- (1,0) [0|65535] "" XXX
 SG_ IS_ACCEL : 5|1@0+ (1,0) [0|1] "" XXX

BO_ 400 UNKNOWN1: 8 XXX
 SG_ ACC_CMD_TRUST : 23|16@0+ (1,0) [0|65535] "" XXX
 SG_ NEW_SIGNAL_1 : 39|16@0+ (1,0) [0|65535] "" XXX
 SG_ CHECKSUM : 63|8@0+ (1,0) [0|255] "" XXX

BO_ 409 UNKNOWN3: 8 XXX
 SG_ LOW_RES_SPEED : 31|8@0+ (2,0) [0|255] "" XXX
 SG_ CHECKSUM : 63|8@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_1 : 47|6@0+ (1,0) [0|63] "" XXX
 SG_ NEW_SIGNAL_2 : 1|1@0+ (1,0) [0|1] "" XXX

BO_ 416 WHEEL_SPEED: 8 XXX
 SG_ CHECKSUM : 63|8@0+ (1,0) [0|255] "" XXX
 SG_ WHEELSPEED_F : 7|24@0+ (0.0000166,0) [0|16777215] "" XXX
 SG_ ACCEL : 31|24@0- (0.000001,0) [0|16777215] "" XXX

BO_ 417 MAYBE_LKAS: 8 XXX
 SG_ NEW_SIGNAL_3 : 24|16@1+ (1,0) [0|65535] "" XXX

BO_ 427 SPEED: 8 XXX
 SG_ SET_ME_1 : 46|1@0+ (1,0) [0|1] "" XXX
 SG_ CHECKSUM : 63|8@0+ (1,0) [0|255] "" XXX
 SG_ WHEELSPEED : 15|16@0+ (0.0108,0) [0|65535] "" XXX

BO_ 448 EPS_SHAFT_TORQUE: 8 XXX
 SG_ CHECKSUM : 31|8@0+ (1,0) [0|255] "" XXX
 SG_ STEERING_TORQUE : 8|8@1+ (1,0) [0|255] "" XXX

BO_ 464 STEERING_LKAS: 8 XXX
 SG_ COUNTER : 8|4@1+ (1,0) [0|15] "" XXX
 SG_ CHECKSUM : 63|8@0+ (1,0) [0|255] "" XXX
 SG_ SET_ME_1 : 22|1@0+ (1,0) [0|1] "" XXX
 SG_ SET_ME_1_2 : 30|1@0+ (1,0) [0|1] "" XXX
 SG_ STEER_CMD : 7|11@0- (1,0) [0|255] "" XXX
 SG_ SET_ME_0 : 39|24@0+ (1,0) [0|16777215] "" XXX
 SG_ STEER_REQ : 21|1@0+ (1,0) [0|1] "" XXX

BO_ 516 RIGHT_STALK: 8 XXX
 SG_ LEFT_SIGNAL : 9|1@0+ (1,0) [0|1] "" XXX
 SG_ RIGHT_SIGNAL : 8|1@0+ (1,0) [0|1] "" XXX
 SG_ GENERIC_TOGGLE : 1|1@0+ (1,0) [0|1] "" XXX
 SG_ HIGH_BEAM : 0|1@0+ (1,0) [0|1] "" XXX

BO_ 520 PCM_BUTTONS: 8 XXX
 SG_ ACC_RDY : 9|1@0+ (1,0) [0|1] "" XXX
 SG_ RES_PLUS : 6|1@0+ (1,0) [0|1] "" XXX
 SG_ SET_MINUS : 5|1@0+ (1,0) [0|1] "" XXX
 SG_ CANCEL : 4|1@0+ (1,0) [0|1] "" XXX
 SG_ COUNTER : 24|4@1+ (1,0) [0|15] "" XXX
 SG_ PEDAL_DEPRESSED : 12|1@0+ (1,0) [0|1] "" XXX
 SG_ CHECKSUM : 40|8@1+ (1,0) [0|255] "" XXX

BO_ 524 TRANSMISSION: 8 XXX
 SG_ CHECKSUM : 47|8@0+ (1,0) [0|255] "" XXX
 SG_ GEAR : 15|4@0+ (1,0) [0|255] "" XXX

BO_ 608 WHEEL_SPEED_CLEAN: 8 XXX
 SG_ CHECKSUM : 63|8@0+ (1,0) [0|255] "" XXX
 SG_ WHEELSPEED : 44|13@0+ (0.01,0) [0|8191] "" XXX

BO_ 624 ADAS_AEB: 8 XXX
 SG_ SET_ME_XC0 : 7|8@0+ (1,0) [0|255] "" XXX
 SG_ SET_ME_XFF_1 : 23|8@0+ (1,0) [0|255] "" XXX
 SG_ SET_ME_XFF_2 : 31|8@0+ (1,0) [0|255] "" XXX
 SG_ SET_ME_XFF_3 : 39|8@0+ (1,0) [0|255] "" XXX
 SG_ SET_ME_XFF_4 : 47|8@0+ (1,0) [0|255] "" XXX
 SG_ SET_ME_X15_1 : 15|8@0+ (1,0) [0|255] "" XXX
 SG_ SET_ME_X15_2 : 55|8@0+ (1,0) [0|255] "" XXX
 SG_ SET_ME_X60 : 63|8@0+ (1,0) [0|255] "" XXX

BO_ 625 ACC_BRAKE: 8 XXX
 SG_ COUNTER : 52|3@0+ (1,0) [0|255] "" XXX
 SG_ CHECKSUM : 63|8@0+ (1,0) [0|255] "" XXX
 SG_ SET_ME_1_WHEN_ENGAGE : 8|1@0+ (1,0) [0|1] "" XXX
 SG_ MAGNITUDE : 47|8@0+ (0.01,-2) [0|255] "" XXX
 SG_ BRAKE_REQ : 13|1@0+ (1,0) [0|1] "" XXX
 SG_ PUMP_REACTION2 : 31|8@0- (0.1,0) [0|16777215] "" XXX
 SG_ PUMP_REACTION1 : 35|4@0+ (0.1,0) [0|1] "" XXX

BO_ 627 ACC_CMD_HUD: 8 XXX
 SG_ CHECKSUM : 63|8@0+ (1,0) [0|255] "" XXX
 SG_ SET_SPEED : 7|8@0+ (1,0) [0|255] "" XXX
 SG_ FOLLOW_DISTANCE : 15|2@0+ (1,0) [0|3] "" XXX
 SG_ IS_LEAD : 11|1@0+ (1,0) [0|1] "" XXX
 SG_ IS_ACCEL : 38|1@0+ (1,0) [0|1] "" XXX
 SG_ IS_DECEL : 37|1@0+ (1,0) [0|1] "" XXX
 SG_ SET_ME_1_2 : 9|1@0+ (1,0) [0|1] "" XXX
 SG_ SET_ME_1 : 54|1@0+ (1,0) [0|1] "" XXX
 SG_ SET_0_WHEN_ENGAGE : 40|1@0+ (1,0) [0|1] "" XXX
 SG_ ACC_CMD : 23|16@0+ (0.0001045,0) [0|127] "" XXX
 SG_ SET_1_WHEN_ENGAGE : 13|1@0+ (1,0) [0|1] "" XXX

CM_ SG_ 627 FOLLOW_DISTANCE "3 is 0, 2 is 1, 1 is 2";

BO_ 628 LKAS_HUD: 8 XXX
 SG_ LKAS_SET : 19|1@1+ (1,0) [0|3] "" XXX
 SG_ LKAS_ENGAGED : 30|1@0+ (1,0) [0|3] "" XXX
 SG_ LANE_RIGHT_DETECT : 34|1@0+ (1,0) [0|7] "" XXX
 SG_ LANE_LEFT_DETECT : 36|1@0+ (1,0) [0|1] "" XXX
 SG_ CHECKSUM : 63|8@0+ (1,0) [0|255] "" XXX
 SG_ SET_ME_X02 : 48|8@1+ (1,0) [0|3] "" XXX
 SG_ LDA_RELATED2 : 18|1@0+ (1,0) [0|1] "" XXX
 SG_ LDA_RELATED1 : 35|1@0+ (1,0) [0|1] "" XXX
 SG_ LDA_ALERT : 14|1@0+ (1,0) [0|1] "" XXX

BO_ 834 BSM: 8 XXX
 SG_ BSM_CHIME : 23|1@0+ (1,0) [0|1] "" XXX

BO_ 856 METER_CLUSTER: 8 XXX
 SG_ SEAT_BELT_WARNING : 58|2@1+ (1,0) [0|3] "" XXX
 SG_ RIGHT_SIGNAL : 15|1@0+ (1,0) [0|3] "" XXX
 SG_ LEFT_SIGNAL : 14|1@0+ (1,0) [0|3] "" XXX
 SG_ MAIN_DOOR : 2|1@1+ (1,0) [0|3] "" XXX
 SG_ LEFT_BACK_DOOR : 5|1@0+ (1,0) [0|1] "" XXX
 SG_ LEFT_FRONT_DOOR : 3|1@0+ (1,0) [0|1] "" XXX
 SG_ RIGHT_BACK_DOOR : 4|1@0+ (1,0) [0|1] "" XXX

BO_ 900 HANDBRAKE: 8 XXX
 SG_ HANDBRAKE_ENGAGED : 24|1@0+ (1,0) [0|1] "" XXX

BO_ 1168 ADAS_ACC: 8 XXX
 SG_ CHECKSUM : 63|8@0+ (1,0) [0|255] "" XXX
 SG_ SET_ME_X18 : 39|8@0+ (1,0) [0|255] "" XXX
 SG_ NEW_SIGNAL_2 : 47|16@0- (1,0) [0|65535] "" XXX
 SG_ NEW_SIGNAL_3 : 23|16@0- (1,0) [0|65535] "" XXX
 SG_ NEW_SIGNAL_1 : 7|16@0- (1,0) [0|4294967295] "" XXX

BO_ 1267 FWD_CAM_HEARTBEAT: 8 CAR
 SG_ SET_ME_X23 : 7|8@0+ (1,0) [0|255] "" ANY
 SG_ SET_ME_X01_1 : 23|8@0+ (1,0) [0|255] "" ANY
 SG_ SET_ME_X01_2 : 31|8@0+ (1,0) [0|255] "" ANY


VAL_ 513 STATE 5 "FAULT_TIMEOUT" 4 "FAULT_STARTUP" 3 "FAULT_SCE" 2 "FAULT_SEND" 1 "FAULT_BAD_CHECKSUM" 0 "NO_FAULT" ;
VAL_ 524 GEAR 8 "R" 4 "N" 2 "D" 0 "P" ;
